nrf-hal-common/src/wdt.rs

```diff
@@ -0,0 +1,450 @@
+        fn create_handle() -> Self::Handles {
+            (
+                WatchdogHandle(Hdl0),
+                WatchdogHandle(Hdl1),
+                WatchdogHandle(Hdl2),
+                WatchdogHandle(Hdl3),
+            )
+        }
+    }
+    impl Handles for Five {
+        type Handles = (
+            WatchdogHandle<Hdl0>,
+            WatchdogHandle<Hdl1>,
+            WatchdogHandle<Hdl2>,
+            WatchdogHandle<Hdl3>,
+            WatchdogHandle<Hdl4>,
+        );
+        const ENABLE: u32 = 0b0001_1111;
+        fn create_handle() -> Self::Handles {
+            (
+                WatchdogHandle(Hdl0),
+                WatchdogHandle(Hdl1),
+                WatchdogHandle(Hdl2),
+                WatchdogHandle(Hdl3),
+                WatchdogHandle(Hdl4),
+            )
+        }
+    }
+    impl Handles for Six {
+        type Handles = (
+            WatchdogHandle<Hdl0>,
+            WatchdogHandle<Hdl1>,
+            WatchdogHandle<Hdl2>,
+            WatchdogHandle<Hdl3>,
+            WatchdogHandle<Hdl4>,
+            WatchdogHandle<Hdl5>,
+        );
+        const ENABLE: u32 = 0b0011_1111;
+        fn create_handle() -> Self::Handles {
+            (
+                WatchdogHandle(Hdl0),
+                WatchdogHandle(Hdl1),
+                WatchdogHandle(Hdl2),
+                WatchdogHandle(Hdl3),
+                WatchdogHandle(Hdl4),
+                WatchdogHandle(Hdl5),
+            )
+        }
+    }
+    impl Handles for Seven {
+        type Handles = (
+            WatchdogHandle<Hdl0>,
+            WatchdogHandle<Hdl1>,
+            WatchdogHandle<Hdl2>,
+            WatchdogHandle<Hdl3>,
+            WatchdogHandle<Hdl4>,
+            WatchdogHandle<Hdl5>,
+            WatchdogHandle<Hdl6>,
+        );
+        const ENABLE: u32 = 0b0111_1111;
+        fn create_handle() -> Self::Handles {
+            (
+                WatchdogHandle(Hdl0),
+                WatchdogHandle(Hdl1),
+                WatchdogHandle(Hdl2),
+                WatchdogHandle(Hdl3),
+                WatchdogHandle(Hdl4),
+                WatchdogHandle(Hdl5),
+                WatchdogHandle(Hdl6),
+            )
+        }
+    }
+    impl Handles for Eight {
+        type Handles = (
+            WatchdogHandle<Hdl0>,
+            WatchdogHandle<Hdl1>,
+            WatchdogHandle<Hdl2>,
+            WatchdogHandle<Hdl3>,
+            WatchdogHandle<Hdl4>,
+            WatchdogHandle<Hdl5>,
+            WatchdogHandle<Hdl6>,
+            WatchdogHandle<Hdl7>,
+        );
+        const ENABLE: u32 = 0b1111_1111;
+        fn create_handle() -> Self::Handles {
+            (
+                WatchdogHandle(Hdl0),
+                WatchdogHandle(Hdl1),
+                WatchdogHandle(Hdl2),
+                WatchdogHandle(Hdl3),
+                WatchdogHandle(Hdl4),
+                WatchdogHandle(Hdl5),
+                WatchdogHandle(Hdl6),
+                WatchdogHandle(Hdl7),
+            )
+        }
+    }
```

Comment:
Do you think you could use a macro to generate these impls? Might make the code a bit shorter.

Reply:
I'm not against it, but I've already typed it out, and I have very little lust auf writing a macro at the moment.